org.openjdk.jmc.console.ext.agent/src/main/java/org/openjdk/jmc/console/ext/agent/manager/wizards/PresetManagerPage.java

```diff
@@ -58,29 +82,177 @@ public PresetManagerPage(PresetRepository repository) {
+		tableViewer.getControl().setLayoutData(new GridData(SWT.FILL, SWT.FILL, true, true));
+		tableViewer.setContentProvider(new PresetTableContentProvider());
+		tableViewer.setLabelProvider(new PresetTableLabelProvider());
+		tableViewer.setInput(repository);
+	}
+
+	private void createButtons(Composite parent) {
+		Composite container = new Composite(parent, SWT.NONE);
+		container.setLayoutData(new GridData(SWT.FILL, SWT.BEGINNING, false, true));
+		GridLayout layout = new GridLayout(1, true);
+		layout.marginHeight = 0;
+		layout.marginWidth = 0;
+		container.setLayout(layout);
+
+		newButton = createButton(container, MESSAGE_NEW_BUTTON);
+		editButton = createButton(container, MESSAGE_EDIT_BUTTON);
+		duplicateButton = createButton(container, MESSAGE_DUPLICATE_BUTTON);
+		removeButton = createButton(container, MESSAGE_REMOVE_BUTTON);
+		importButton = createButton(container, MESSAGE_IMPORT_FILES_BUTTON);
+		exportButton = createButton(container, MESSAGE_EXPORT_FILE_BUTTON);
+	}
+
+	private Button createButton(Composite parent, String text) {
+		Button button = new Button(parent, SWT.NONE);
+		button.setText(text);
+		button.setLayoutData(new GridData(SWT.FILL, SWT.FILL, true, false));
+		return button;
 	}
 
 	// TODO: call this function when "New" or "Edit" button clicked
 	private void openPresetEditingWizardFor(IPreset preset) {
-		if (DialogToolkit.openWizardWithHelp(new PresetEditingWizard(preset))) {
-			// TODO: save the modified preset to the repository
+		if (!DialogToolkit.openWizardWithHelp(new PresetEditingWizard(preset))) {
+			return;
+		}
+
+		// TODO: save the modified preset to the repository 
+		repository.add(preset);
+	}
+
+	private void bindListeners() {
+		// TODO: should the repository be Observable and refresh the view when it changes?
+		newButton.addSelectionListener(new SelectionAdapter() {
+			@Override
+			public void widgetSelected(SelectionEvent e) {
+				openPresetEditingWizardFor(new Preset());
+				tableViewer.refresh();
+			}
+		});
+
+		editButton.addSelectionListener(new SelectionAdapter() {
+			@Override
+			public void widgetSelected(SelectionEvent e) {
+				openPresetEditingWizardFor((IPreset) tableViewer.getStructuredSelection().getFirstElement());
+				tableViewer.refresh();
+			}
+		});
+		editButton.setEnabled(false);
+
+		duplicateButton.addSelectionListener(new SelectionAdapter() {
+			@Override
+			public void widgetSelected(SelectionEvent e) {
+				// TODO: create a copy properly
+				IPreset original = (IPreset) tableViewer.getStructuredSelection().getFirstElement();
+				IPreset duplicate = new Preset();
+				duplicate.setFileName("Copy of " + original.getFileName());
+				repository.add(duplicate);
+				tableViewer.refresh();
+			}
+		});
+		duplicateButton.setEnabled(false);
+
+		importButton.addSelectionListener(new SelectionAdapter() {
+			@Override
+			public void widgetSelected(SelectionEvent e) {
+				// TODO: file selection dialog
```

Comment:
Is this supposed to be done in this PR or a future one? Same with the export button.